### PR TITLE
Chat messages with no language are now handled correctly.

### LINKED ISSUE #6111: Removing a xenomorph's alien tongue allows them to speak Common in runechat
## Testmerges:

KapuLimbs I think is being testmerged still, I can't see how I can check since my chat window has already scrolled past the beginning logs.

## Reproduction:

A human scientist was turned into a xenomorph I believe (Xul-Ru), and I observed a crew member operating on them to remove their alien tongue. After removal, when they tried to speak it showed in the chat log it showed as "Xul-Ru makes a strange sound." but it showed as Galactic Common in runechat above their head. They weren't able to understand Common themselves, but other crew members could understand them.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
Fixes #6111

## About The Pull Request

Overhead chat messages created with a null language now are replaced with *makes a strange sound emote.

## Why It's Good For The Game

Fixes a bug that meant if someone speaks without a languge

image

## Changelog
:cl:
fix: Runechat now properly handles null languages.
/:cl:

### CODE REVIEW

**Review by ike709 — DISMISSED** [review]

File: code/datums/chatmessage.dm [hunk 1]
```diff
@@ -313,7 +313,9 @@ GLOBAL_LIST_INIT(job_colors_pastel, list(
 		if(isAI(src) && !GLOB.cameranet.checkCameraVis(v.source))
 			return CHATMESSAGE_CANNOT_HEAR
 	var/datum/language/language_instance = GLOB.language_datum_instances[message_language]
-	if(language_instance?.display_icon(src))
+	if(!language_instance)
+		return CHATMESSAGE_HEAR
+	if(language_instance.display_icon(src))
 		return CHATMESSAGE_SHOW_LANGUAGE_ICON
 	return CHATMESSAGE_HEAR
```
Comment (Kapu1178) [inline code/datums/chatmessage.dm]: ...Take a second look at this and tell me whats wrong.

**Comment by Kapu1178** [comment]
tg fixed this issue in a more graceful way. hijacking raw_message is icky and gross. https://github.com/tgstation/tgstation/pull/54318

File: code/datums/chatmessage.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/datums/chatmessage.dm]: Nothing

**Comment by PowerfulBacon** [comment]
> tg fixed this issue in a more graceful way. hijacking raw_message is icky and gross. [tgstation/tgstation#54318](https://github.com/tgstation/tgstation/pull/54318)

I mean TGs one hijacks raw message anyway by using lang_treat. This just adds in the only part of lang_treat that hasn't been implemented which is the makes a strange sound check for if you speak with no languages available. The reason I don't use lang_treat is because lang_treat treats the language for a specific mob, but we need to treat it for a generic case to create the messages in a more performant manner.

File: code/datums/chatmessage.dm [hunk 2]
```diff
@@ -313,7 +313,9 @@ GLOBAL_LIST_INIT(job_colors_pastel, list(
 		if(isAI(src) && !GLOB.cameranet.checkCameraVis(v.source))
 			return CHATMESSAGE_CANNOT_HEAR
 	var/datum/language/language_instance = GLOB.language_datum_instances[message_language]
-	if(language_instance?.display_icon(src))
+	if(!language_instance)
+		return CHATMESSAGE_HEAR
+	if(language_instance.display_icon(src))
```
Comment (Archanial) [inline code/datums/chatmessage.dm]: this does not change anything

Comment (PowerfulBacon) [reply code/datums/chatmessage.dm]: maybe but I can't think if it does or not

Comment (PowerfulBacon) [inline code/datums/chatmessage.dm]:
```suggestion
	if(language_instance?.display_icon(src))
```

**Review by Archanial — APPROVED** [review]

**Pull request merged**